.github/workflows/test.yml

```diff
@@ -25,6 +25,7 @@ env:
   VALGRIND_LOG: ${{ github.workspace }}/build/log/valgrind-%p.log
   # TEST_FILE: test/functional/core/startup_spec.lua
   # TEST_FILTER: foo
+  TEST_INTEG: true
```

Comment:
I think only 1 CI job should enable this. Also, `NVIM_TEST_INTEG` was introduced [recently](https://github.com/neovim/neovim/blob/8a7bfdc0792229e87fcaebdcf2e877fb91e2455c/test/README.md?plain=1#L524) (but not enabled in CI), so let's use that (I don't have a preference on the name though).

Reply:
I'd think that at least testing on sufficiently different environments is worth it. For example: basic (Linux+LuaJIT), MacOS, Windows, PUC Lua. This would require some magic in yaml files, though. Does that sound good?

Reply:
[Enabled on release build](https://github.com/neovim/neovim/pull/35052/commits/9b486f6f55538a20021c5e8b81d54c576210365c).